api/src/main/java/me/lokka30/treasury/api/economy/account/accessor/NonPlayerAccountAccessor.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/lokka30/Treasury>.
+ */
+
+package me.lokka30.treasury.api.economy.account.accessor;
+
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import me.lokka30.treasury.api.common.response.Response;
+import me.lokka30.treasury.api.economy.account.NonPlayerAccount;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Represents an account accessor, which accesses {@link NonPlayerAccount non player accounts}
+ *
+ * @author MrIvanPlays
+ * @since 2.0.0
+ */
+public abstract class NonPlayerAccountAccessor {
+
+    private String identifier, name;
+
+    protected NonPlayerAccountAccessor() {
+    }
+
+    /**
+     * Specifies the identifier of the accessed/created {@link NonPlayerAccount non player account}
+     *
+     * @param identifier account identifier
+     * @return this instance for chaining
+     * @since 2.0.0
+     */
+    @NotNull
+    public NonPlayerAccountAccessor withIdentifier(@NotNull String identifier) {
+        this.identifier = Objects.requireNonNull(identifier, "identifier");
+        return this;
+    }
+
+    /**
+     * Specifies the name of the accessed/created {@link NonPlayerAccount non player account}.
+     * This is optional to set.
+     *
+     * @param name account name
+     * @return this instance for chaining
+     * @since 2.0.0
+     */
+    @NotNull
```

Comment:
We should note the purpose of this method: as this method can create a non-player account, it may be desired to give the account a specific name alongside its identifier.

Side-thought: this method could be removed and providers can separately provide the name for the non-player account, after it has been created+retrieved.

Reply:
can't come up with anything about how this method shall be clarified. i also don't think it shall be removed.

Reply:
Actually, I have decided that this method shouldn't be here. It doesn't serve as primary identification for the non-player account, it doesn't belong there.

Reply:
sure but then that's making 2 database queries rather than 1, why not have it directly specified?

Reply:
I'm on the fence about it, I'd like an additional person's opinion on it. 🙂

Reply:
@yannicklamprecht @Jikoo opinion will be appreciated.

Reply:
Let's just keep it this way and merge the PR